rules/TypeDeclaration/Rector/ClassMethod/AddParamTypeBasedOnPHPUnitDataProviderRector.php

```diff
@@ -353,12 +353,11 @@ private function refactorClassMethod(ClassMethod $classMethod, Class_ $class, ar
                 continue;
             }
 
-            $param->type = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode(
-                $paramTypeDeclaration,
-                TypeKind::PARAM
-            );
-
-            $hasChanged = true;
+            $type = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode($paramTypeDeclaration, TypeKind::PARAM);
+            if ($type instanceof Node) {
+                $param->type = $type;
```

Comment:
We only need to set this as changed if we actually change the param type

Reply:
Thanks :+1: